pkg/scalers/gcp_storage_scaler.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+
+	bucket := client.Bucket(meta.bucketName)
+	if bucket == nil {
+		return nil, fmt.Errorf("failed to create a handle to bucket %s", meta.bucketName)
+	}
+
+	gcsLog.Info(fmt.Sprintf("Metadata %v", meta))
+
+	return &gcsScaler{
+		client:   client,
+		bucket:   bucket,
+		metadata: meta,
+	}, nil
+}
+
+func parseGcsMetadata(config *ScalerConfig) (*gcsMetadata, error) {
+	meta := gcsMetadata{}
+	meta.targetObjectCount = defaultTargetObjectCount
+	meta.maxBucketItemsToScan = defaultMaxBucketItemsToScan
+
+	if val, ok := config.TriggerMetadata["bucketName"]; ok {
+		if val == "" {
+			gcsLog.Error(nil, "no bucket name given")
+			return nil, fmt.Errorf("no bucket name given")
+		}
+
+		meta.bucketName = val
+	} else {
+		gcsLog.Error(nil, "no bucket name given")
+		return nil, fmt.Errorf("no bucket name given")
+	}
+
+	if val, ok := config.TriggerMetadata["targetObjectCount"]; ok {
+		targetObjectCount, err := strconv.Atoi(val)
+		if err != nil {
+			gcsLog.Error(err, "Error parsing targetObjectCount")
+			return nil, fmt.Errorf("error parsing targetObjectCount: %s", err.Error())
+		}
+
+		meta.targetObjectCount = targetObjectCount
+	}
+
+	if val, ok := config.TriggerMetadata["maxBucketItemsToScan"]; ok {
+		maxBucketItemsToScan, err := strconv.Atoi(val)
+		if err != nil {
+			gcsLog.Error(err, "Error parsing maxBucketItemsToScan")
+			return nil, fmt.Errorf("error parsing maxBucketItemsToScan: %s", err.Error())
+		}
+
+		meta.maxBucketItemsToScan = maxBucketItemsToScan
+	}
+
+	if val, ok := config.TriggerMetadata["endpoint"]; ok {
+		if val != "" {
+			meta.endpoint = val
+		}
+	}
+
+	if meta.endpoint == "" {
+		auth, err := getGcpAuthorization(config, config.ResolvedEnv)
+		if err != nil {
+			return nil, err
+		}
+		meta.gcpAuthorization = auth
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("gcp-storage-%s", val))
+	} else {
+		meta.metricName = kedautil.NormalizeString(fmt.Sprintf("gcp-storage-%s", meta.bucketName))
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, nil
+}
+
+// IsActive checks if there are any messages in the subscription
+func (s *gcsScaler) IsActive(ctx context.Context) (bool, error) {
+	items, err := s.getItemCount(ctx, 1)
+	if err != nil {
+		return false, err
+	}
+
+	return items > 0, nil
+}
+
+func (s *gcsScaler) Close(context.Context) error {
+	if s.client != nil {
+		return s.client.Close()
+	}
+	return nil
+}
+
+// GetMetricSpecForScaling returns the metric spec for the HPA
+func (s *gcsScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	targetValueQty := resource.NewQuantity(int64(s.metadata.targetObjectCount), resource.DecimalSI)
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, s.metadata.metricName),
```

Comment:
Nit: could you please move this out of this function, so it is not called every time: 

You can check this example: https://github.com/kedacore/keda/blob/b6bc7b754e2b5145c45895126e9dc57e86ab8816/pkg/scalers/datadog_scaler.go#L134